dfc/target.go

```diff
@@ -426,7 +426,8 @@ existslocally:
 	written, err := io.CopyBuffer(w, file, buf)
 	if err != nil {
 		errstr = fmt.Sprintf("Failed to send file %s, err: %v", fqn, err)
-		t.invalmsghdlr(w, r, errstr)
+		glog.Errorln(t.richHTTPError(r, errstr, http.StatusInternalServerError))
+		t.statsif.add("numerr", 1)
```

Comment:
this means each call to richHTTPError() has to call this?
as i mentioned last week, what i was thinking is do something like this:
fun nameForNewErrorFunc(w, r, errstr, status, ) {
   does everything existing invalmsghdlr() does but without the part that collect stats
	if len(other) > 0 {
 		status = other[0].(int)
 	}
}

so existing invalmsghdlr() becomes:
{ 
   extract status and call nameForNewErrorFunc()
}

and your new fund simply call nameForNewErrorFunc() with the right status and error